src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -399,6 +400,17 @@ static bool HasNonWhitespace(ReadOnlySpan<byte> span)
         return result;
     }
 
+    private bool _hasWarnedAbout4444 = false;
+    private bool ShouldWarn(int code)
```

Comment:
not sure, but maybe drop warnings only on engine API endpoint and keep them on standard RPC endpoint?